tracking: failing ef-tests

```
FAILED cairo/tests/ef_tests/cancun/test_state_transition.py::test_general_state_tests[bcWalletTest - walletReorganizeOwners_Cancun] - Failed: Timeout >300.0s
```

List of failing ef-tests. Investigate & fix them.

--- a/cairo/ethereum/cancun/vm/gas.cairo
+++ b/cairo/ethereum/cancun/vm/gas.cairo
@@ -281,16 +281,16 @@ func calculate_total_blob_gas{range_check_ptr}(tx: Transaction) -> Uint {
     return total_blob_gas;
 }
 
-// @dev: Saturates at 2**64 - 1
+// @dev: Saturates at 2**128 - 1
 func calculate_blob_gas_price{range_check_ptr}(excess_blob_gas: U64) -> Uint {
     let blob_gas_price = taylor_exponential(
         Uint(GasConstants.MIN_BLOB_GASPRICE),
         Uint(excess_blob_gas.value),
         Uint(GasConstants.BLOB_GASPRICE_UPDATE_FRACTION),
     );
-    let saturate = is_le_felt(2 ** 64, blob_gas_price.value);
+    let saturate = is_le_felt(2 ** 128, blob_gas_price.value);
     if (saturate != 0) {
-        tempvar res = Uint(2 ** 64 - 1);
+        tempvar res = Uint(2 ** 128 - 1);
         return res;
     }
     return blob_gas_price;
@@ -300,9 +300,14 @@ func calculate_data_fee{range_check_ptr}(excess_blob_gas: U64, tx: Transaction) 
     alloc_locals;
     // saturate at 2**64 - 1
     let total_blob_gas = calculate_total_blob_gas(tx);
-    // saturate at 2**64 - 1
+    // saturate at 2**128 - 1
     let blob_gas_price = calculate_blob_gas_price(excess_blob_gas);
-    // fits in (2**128-1)
+    // mul fits in 251 bits
+    let saturate = is_le_felt(2 ** 128, total_blob_gas.value * blob_gas_price.value);
+    if (saturate != 0) {
+        tempvar res = Uint(2 ** 128 - 1);
+        return res;
+    }
     let data_fee = Uint(total_blob_gas.value * blob_gas_price.value);
     return data_fee;
 }

--- a/cairo/ethereum/cancun/vm/instructions/environment.cairo
+++ b/cairo/ethereum/cancun/vm/instructions/environment.cairo
@@ -927,6 +927,7 @@ func blob_base_fee{
 
     let _blob_base_fee = calculate_blob_gas_price(evm.value.env.value.excess_blob_gas);
 
+    // Result saturated to fit in 128 bits
     tempvar blob_base_fee = U256(new U256Struct(_blob_base_fee.value, 0));
     let stack = evm.value.stack;
     with stack {

--- a/cairo/ethereum_rlp/rlp.cairo
+++ b/cairo/ethereum_rlp/rlp.cairo
@@ -1069,7 +1069,7 @@ func decode_to_blob_transaction{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(
     let items = decoded.value.sequence.value.data;
 
     // EIP-4844 transactions must have 14 fields (11 transaction fields + 3 signature fields)
-    with_attr error_message("Invalid blob transaction: wrong number of fields") {
+    with_attr error_message("DecodingError") {
         assert items_len = 14;
     }
 

--- a/cairo/tests/ef_tests/helpers/load_state_tests.py
+++ b/cairo/tests/ef_tests/helpers/load_state_tests.py
@@ -125,10 +125,12 @@ def add_block_to_chain(
         chain.blocks = cairo_chain.blocks
         chain.state = cairo_chain.state
     except Exception as e:
-        # Run EELS to get its trace, then raise.
+        # In trace mode, run EELS as well to get a side-by-side comparison
         if request.config.getoption("--log-cli-level") == "TRACE":
-            # In trace mode, run EELS as well to get a side-by-side comparison
-            state_transition(chain, block)
+            try:
+                state_transition(chain, block)
+            except Exception:
+                pass
         raise e
 
 

--- a/python/cairo-addons/src/cairo_addons/testing/hooks.py
+++ b/python/cairo-addons/src/cairo_addons/testing/hooks.py
@@ -315,7 +315,7 @@ def pytest_collection_modifyitems(session, config, items):
             ]
         if (
             config.getoption("skip_cached_tests")
-            and session.test_hashes[item.nodeid] in tests_to_skip
+            and session.test_hashes.get(item.nodeid) in tests_to_skip
         ):
             item.add_marker(pytest.mark.skip(reason="Cached results"))
 

